### PR TITLE
Use a custom container image optimised for size

### PR DESCRIPTION
**Pull request by betatim**
This is an experiment. I am not sure this is a good idea or if it will even achieve its goal.

This PR replaces to `requirements.txt` with a `Dockerfile` which is copied from https://github.com/binder-examples/minimal-dockerfile. The goal of that repository is to show people what the absolute minimal requirements are for a `Dockerfile` based container to work on a BinderHub. I've taken that and added the dependencies of this repository to it.

It produces a very small image which should in theory/one could hope be faster to launch than a normal image built by repo2docker's magic. Smaller usually means faster (at the expense that you now need to maintain a `Dockerfile` which isn't a lot of fun) but it isn't clear cut because docker images can share a lot with other images. So there is a benefit to using parts of a popular image (like we do when using repo2docker) and only bringing in the last few changed layers which contain the customisations of this repository. This means it could be that the default way (not this PR) ends up being quicker in practice than this approach of using a custom image.

I am not sure how to nicely measure the time difference between using this image and the original one. Still thinking on that.

### CODE REVIEW

**Comment by hfboyce** [comment]
@betatim I'm sorry about the delay. I am so absolutely grateful for this PR. The slow loading environment have been a constant issue for us and the fact you took time out of your day to help us attempt to figure this out is so appreciated. 
I have a meeting planned where we will attempt a "before and after" timing of loading the environment (aka the time it appears to the user) and we will continue your experiment and give you results. Stayed tuned! 
Again thank you for contributing to our project!

**Comment by betatim** [comment]
It looks like this got merged into the wrong branch :( At least when I try the website it is still launching from the `binder` branch.

Should I make a new PR?

**Comment by hfboyce** [comment]
No No I can merged the binder branch to master. let's see if this works now!

**Comment by hfboyce** [comment]
There is good news and bad news.  The good news is this works on questions that do not call in datasets! The bad news is the data is not in the binder so questions depending on them don't work now.

image


 I understand that adding the data will contribute to the slower speed but we need to include them. Is there a way to include the data in the data folder?

File: binder/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+                           "wasabi>=0.2.1,<1.1.0" \
+                           "pytest>=4.4.1,<4.5.0" \
+                           "black==19.10b0" \
+                           "altair>=4.1.0" \
+                           "xlrd>=1.0.0" \
+                           "pandas==1.0.5" \
+                           "plotly==4.8.1" \
+                           "numpy>=1.11.0,<1.16.4" \
+                           "matplotlib==3.*"
+
```
Comment (hfboyce) [inline binder/Dockerfile:14]: add line for cloning repo
```
RUN git clone "https://github.com/UBC-MDS/MCL-DSCI-011-programming-in-python.git"
```

**Pull request merged**